### PR TITLE
[Storage] When download blob with parameter AbsoluteUri, not allow in…

### LINKED ISSUE #26316: Get-AzStorageBlobContent download from uri without sas and access key
### Description

In azure cli, it is possible to download a blob content by providing blob uri without sas key or access token
`az storage blob download --blob-url https://<storageaccount>.blob.core.windows.net/<container>/<blob>--auth-mode login`

Is it possible to do the same in az powershell module?

I'm trying this 
```powershell
Connect-AzAccount
$Context = New-AzStorageContext -StorageAccountName <storageaccount> -UseConnectedAccount
Get-AzStorageBlobContent -Uri https://<storageaccount>.blob.core.windows.net/<container>/<blob> -Destination <dst> -Context $Context
```
But it gave me this
```powershell
Get-AzStorageBlobContent -Uri https://<storageaccount>.blob.core.windows.net/<container>/<blob> -Destination <dst> -Context $Context
DEBUG: 4:29:29 PM - Init Operation Context for 'GetAzureStorageBlobContentCommand' with client request id  <>. If you want to get more details, please add "-Debug" to your command.
DEBUG: 4:29:29 PM - [ConfigManager] Got nothing from [EnableErrorRecordsPersistence], Module = [], Cmdlet = []. Returning default value [False].
Get-AzStorageBlobContent: Service request failed.
Status: 401 (Server failed to authenticate the request. Please refer to the information in the www-authenticate header.)
ErrorCode: NoAuthenticationInformation

Headers:
Transfer-Encoding: chunked
Server: Windows-Azure-Blob/1.0,Microsoft-HTTPAPI/2.0
x-ms-request-id: <>
x-ms-client-request-id: <>
x-ms-version: 2023-08-03
x-ms-error-code: NoAuthenticationInformation
WWW-Authenticate: Bearer authorization_uri=https://login.microsoftonline.com/<>/oauth2/authorize resource_id=https://storage.azure.com
Date: Mon, 14 Oct 2024 08:29:28 GMT
DEBUG: 4:29:29 PM - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: 4:29:29 PM - GetAzureStorageBlobContentCommand end processing, Start 6 remote calls. Finish 6 remote calls. Elapsed time 11325099.66 ms. Client operation id: Azure-Storage-PowerShell-ae82538b-d3c6-456d-b4e8-c3b04c75878f.
DEBUG: 4:29:29 PM - [ConfigManager] Got nothing from [DisplayRegionIdentified], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: 4:29:29 PM - GetAzureStorageBlobContentCommand end processing.
```

### Script or Debug output

_No response_

### Environment data
```powershell 
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.5
PSEdition                      Core
GitCommitId                    7.4.5
OS                             Microsoft Windows 10.0.26100
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
### Module versions
```powershell
Get-Module Az*

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     3.0.4                 Az.Accounts                         {Add-AzEnvironment, Clear-AzConfig, Clear-AzContext, Clear-AzDefault…}
Script     7.3.0                 Az.Storage                          {Add-AzRmStorageContainerLegalHold, Add-AzStorageAccountManagementPolicyAction, Add-AzStorage…
```

### Error output

_No response_

### PR DESCRIPTION
**Pull request by VeryEarly**
…put parameter Context (#26585)

* [Storage] fix #26316: When download blob with parameter AbsoluteUri (alias Uri, BlobUri), not allow input parameter Context together.

* Add static exception

* revise change log

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage.Management/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* When download blob with parameter AbsoluteUri (alias Uri, BlobUri), not allow input parameter Context together.
```
Comment (blueww) [inline src/Storage/Storage.Management/ChangeLog.md:21]: Just to confirm after 11/11 release is out, this change log for 11/19 and below change log for 11/11 will be in correct place in change log file.

File: tools/StaticAnalysis/Exceptions/Az.Storage/BreakingChangeIssues.csv [hunk 2]
```diff
@@ -1,2 +1,83 @@
...
+"Az.Storage","Microsoft.WindowsAzure.Commands.Storage.File.Cmdlet.GetAzureStorageShareStoredAccessPolicy","Get-AzStorageShareStoredAccessPolicy","0","1020","The cmdlet 'Get-AzStorageShareStoredAccessPolicy' no longer has output type 'Microsoft.Azure.Storage.File.SharedAccessFilePolicy'.","Make cmdlet 'Get-AzStorageShareStoredAccessPolicy' return type 'Microsoft.Azure.Storage.File.SharedAccessFilePolicy'."
+"Az.Storage","Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet.GetAzureStorageBlobContentCommand","Get-AzStorageBlobContent","0","1050","The parameter set '__AllParameterSets' for cmdlet 'Get-AzStorageBlobContent' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Get-AzStorageBlobContent'."
+"Az.Storage","Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet.GetAzureStorageBlobContentCommand","Get-AzStorageBlobContent","0","1050","The parameter set 'UriPipeline' for cmdlet 'Get-AzStorageBlobContent' has been removed.","Add parameter set 'UriPipeline' back to cmdlet 'Get-AzStorageBlobContent'."
```
Comment (blueww) [inline tools/StaticAnalysis/Exceptions/Az.Storage/BreakingChangeIssues.csv:83]: The exception in line 82-83 is from my change, others are from Yifan's change https://github.com/Azure/azure-powershell/pull/26594.
Just to make sure the PR won't take conflict.

File: src/Storage/Storage.Management/ChangeLog.md (hunk 1 above)
Comment (VeryEarly) [reply src/Storage/Storage.Management/ChangeLog.md:21]: yes you are right, after release branch merged back to main branch

File: tools/StaticAnalysis/Exceptions/Az.Storage/BreakingChangeIssues.csv (hunk 2 above)
Comment (VeryEarly) [reply tools/StaticAnalysis/Exceptions/Az.Storage/BreakingChangeIssues.csv:4]: nice catch, let me include only your changes

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**